### PR TITLE
Loss optimization

### PR DESCRIPTION
**Pull request by siebenkopf**
The objectosphere loss requires a batchsize to be specified, for no reason. 

Here is a simplification, which is backward compatible.

### CODE REVIEW

File: vast/eval/eval.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import torch
-from vast.tools import device, set_device_cpu
+from ..tools import device, set_device_cpu
```
Comment (akshay-raj-dhamija) [inline vast/eval/eval.py:2]: Thanks! I had posted a question on a similar change https://github.com/Vastlab/vast/pull/47#discussion_r753429801 but it probably slipped through your notifications.
I also found that PEP8 recommends absolute imports. For my understanding please let me know your thoughts about the linked comment.

File: vast/losses/losses.py [hunk 2]
```diff
@@ -137,30 +137,27 @@ def __call__(self, logit_values, target, sample_weights=None):
 
 
 class objectoSphere_loss:
-    def __init__(self, batch_size, knownsMinimumMag=50.0):
+    def __init__(self, batch_size = None, knownsMinimumMag=50.0): # batchsize ignored for backward compatibility
```
Comment (akshay-raj-dhamija) [inline vast/losses/losses.py:140]: I think we do not need backward compatibility here since this feature isn't used by many so just have to change my own code.

File: vast/losses/losses.py [hunk 3]
```diff
@@ -137,30 +137,27 @@ def __call__(self, logit_values, target, sample_weights=None):
...
         known_indexes = target != -1
         unknown_indexes = ~known_indexes
+        # knowns: punish if magnitude is inside of ring
         loss[known_indexes] = mag_diff_from_ring[known_indexes]
+        # unknowns: punish any magnitude
         loss[unknown_indexes] = mag[unknown_indexes]
         loss = torch.pow(loss, 2)
         if sample_weights is not None:
             loss = sample_weights * loss
+        # todo: return average loss
```
Comment (akshay-raj-dhamija) [inline vast/losses/losses.py:160]: The reason all loss functions inside vast are designed to return a per sample loss is so that if someone needs to do hard negative mining based on per sample loss they can do it outside without modifying the code here. Though, I do see the need for a reduction function like in most loss functions in torch, which can make this much easier. Will look into this.

File: vast/losses/losses.py (hunk 2 above)
Comment (akshay-raj-dhamija) [reply vast/losses/losses.py]: I have removed the `batch_size` here and in the `MNIST_Experiments` repo.

File: vast/losses/losses.py (hunk 3 above)
Comment (akshay-raj-dhamija) [reply vast/losses/losses.py]: I have made the required changes, now for any function, you can pass a `reduction` argument with choices `none`, `sum` or `mean`.

Comment (siebenkopf) [reply vast/losses/losses.py]: Great, thanks. Do you think, this one can be merged now?

Comment (akshay-raj-dhamija) [reply vast/losses/losses.py]: I am merging it for now, but should likely get back to it at some point. The way I implemented `reduction` it is expected as a parameter in the forward pass rather than being set in the class constructor. The easiest way would be through a generalized loss class and using it as a parent for all classes. I am open to suggestions but merging this for now since it doesn't break anything.

**Review by akshay-raj-dhamija — APPROVED** [review]
